### PR TITLE
cluster-side filtering of metadata keys.  Still done once the final d…

### PR DESCRIPTION
**Pull request by dannon**
…ict is assembled, to allow for all three passes to work correctly with their dependent keys, etc.

ping @natefoo, @blankenberg because I'd really like extra eyes on this one.

### CODE REVIEW

File: lib/galaxy/jobs/__init__.py [hunk 1]
```diff
@@ -1597,6 +1597,7 @@ def setup_external_metadata( self, exec_dir=None, tmp_dir=None,
                                                                       config_file=config_file,
                                                                       datatypes_config=datatypes_config,
                                                                       job_metadata=os.path.join( self.working_directory, TOOL_PROVIDED_JOB_METADATA_FILE ),
+                                                                      max_metadata_value_size = self.app.config.max_metadata_value_size,
```
Comment (nsoranzo) [inline lib/galaxy/jobs/__init__.py]: flake8 is complaining about the extra whitespaces around `=`.

Comment (dannon) [inline lib/galaxy/jobs/__init__.py]: Yep, fixing.  I had relaxed my local settings before we implemented all this, need to update it :)

**Comment by nsoranzo** [comment]
I guess that something like `max_metadata_value_size=app.config.max_metadata_value_size,` should also be added at https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/actions/metadata.py#L75 .

**Comment by dannon** [comment]
@nsoranzo Good idea, might as well -- I was just focusing on the case for new metadata, but it should also be updated.

**Comment by nsoranzo** [comment]
:+1:

**Comment by natefoo** [comment]
This looks great. We'll also need to put this on stable/15.05. Thanks @dannon.

**Comment by natefoo** [comment]
+1, btw.

**Comment by dannon** [comment]
Cool, merging and I'll apply to 15.05, thanks for the review @nsoranzo and @natefoo

**Comment by dannon** [comment]
(done)

**Pull request merged**